addons/mail/static/src/model/model_manager.js

```diff
@@ -653,23 +684,50 @@ export class ModelManager {
             throw Error(`Cannot delete already deleted record ${record.localId}.`);
         }
         record._willDelete();
+        for (const listener of record.__listeners) {
+            this.removeListener(listener);
+        }
         for (const field of Model.__fieldList) {
             if (field.fieldType === 'relation') {
                 // ensure inverses are properly unlinked
                 field.parseAndExecuteCommands(record, unlinkAll(), { allowWriteReadonly: true });
             }
         }
-        this._hasAnyChangeDuringCycle = true;
-        // TODO ideally deleting the record should be done at the top of the
-        // method, and it shouldn't be needed to manually remove
-        // _toComputeFields, but it is not possible until related are also
-        // properly unlinked during `set`.
         this._createdRecords.delete(record);
-        this._toComputeFields.delete(record);
-        this._toCallOnChange.delete(record);
+        // Mark listeners to notify.
+        for (const listener of this._listenersUsingLocalId.get(record.localId)) {
+            this._markListenerToNotify(listener);
+            this._localIdsUsedByListener.get(listener).delete(record.localId);
+        }
+        for (const listener of this._listenersUsingModelAll.get(Model)) {
+            this._markListenerToNotify(listener);
+        }
         delete Model.__records[record.localId];
     }
 
+    /**
+     * Disables the given listener.
+     *
+     * @param {*} listener
+     */
+    _disableListener(listener) {
```

Comment:
Would combine `removeListener` and `_disableListener` together. There's no reason to have both methods when they are pretty much aliases. Also "remove" and "disable" is not synonymous, so I would prefer being consistent and keep only 1 of them.

Reply:
ok